cmake/modules/FindTargetTools.cmake

```diff
@@ -92,6 +92,8 @@ endif()
 # re-set it.
 unset(CMAKE_C_COMPILER)
 unset(CMAKE_C_COMPILER CACHE)
+unset(CMAKE_CXX_COMPILER)
+unset(CMAKE_CXX_COMPILER CACHE)
```

Comment:
A toolchain's `generic.cmake` is intended to provide a C compiler which can be used for pre-processing.

Thus i'm a little curious as to why `CMAKE_CXX_COMPILER` is set by any `generic.cmake` toolchain file.
I see such a change to LLVM generic.cmake made it in here: https://github.com/zephyrproject-rtos/zephyr/commit/fae9923ff2dedafc92dff6f79fff0c1ff50ff742
but I wonder if that itself is a mistake.

@yperess do you recall why `compiler/clang/generic.cmake` sets CMAKE_CXX_COMPILER in #52005 ?
To me that change looks wrong, so I hope you can explain the reason.

Reply:
@tejlmand I can't seem to find any notes about why I made the change in `generic.cmake`, I do seem to recall that I tried moving it around at the time and the only place that stuck was in the generic file.

Reply:
There's no hard requirement for it to be in `generic.cmake`